### PR TITLE
[OCPBUGS36313] Align IBM Z parm files

### PR DESCRIPTION
**Pull request by SNiemann15**
Version(s): 4.16+

Issue: https://issues.redhat.com/browse/OCPBUGS-36313

Link to docs preview:

QE review:
- [x] QE has approved this change.

Additional information:
https://docs.google.com/document/d/1Vfw2PHpyWFk61xL19hbuN1b1Pc-HCTHPKRCJddprYLE/edit#heading=h.uwdou3t3oz3l

### CODE REVIEW

**Review by smolin-de — COMMENTED** [review]
These are my initial findings.
I need to check some parameters, therefore I do not approve now this PR.
Additional changes may be required.

File: modules/install-sno-ibm-z.adoc [hunk 1]
```diff
@@ -150,20 +150,20 @@ The `rootfs` image is the same for FCP and DASD.
...
-rd.neednet=1 \
+cio_ignore=all,!condev rd.neednet=1 \
 console=ttysclp0 \
-coreos.live.rootfs_url=<rhcos_liveos>:8080/rootfs.img \// <1>
 ignition.firstboot ignition.platform.id=metal \
-ignition.config.url=<rhcos_ign>:8080/ignition/bootstrap-in-place-for-live-iso.ign \// <2>
-ip=encbdd0:dhcp::02:00:00:02:34:02 <3>
+ignition.config.url=http://<http_server>:8080/ignition/bootstrap-in-place-for-live-iso.ign \// <1>
+coreos.live.rootfs_url=http://<http_server>/rhcos-<version>-live-rootfs.<architecture>.img \// <2>
+ip=<ip>::<gateway>:<mask>:<hostname>::none <3>
```
Comment (smolin-de) [inline modules/install-sno-ibm-z.adoc:158]:
```suggestion
ip=<ip>::<gateway>:<mask>:<hostname>::none nameserver=<dns> <3>
```

File: modules/install-sno-ibm-z-kvm.adoc [hunk 2]
```diff
@@ -159,15 +159,16 @@ $ virt-install \
...
-   --extra-args "ip=<ip>::<gateway>:<mask>:<hostname>::none" \
-   --extra-args "nameserver=<name_server>" \
    --extra-args "ip=dhcp rd.neednet=1 ignition.platform.id=metal ignition.firstboot" \
-   --extra-args "coreos.live.rootfs_url=<rhcos_liveos>" \// <2>
-   --extra-args "ignition.config.url=<rhcos_ign>" \// <3>
+   --extra-args "ignition.config.url=http://<http_server>/bootstrap.ign" \// <2>
+   --extra-args "coreos.live.rootfs_url=http://<http_server>/rhcos-<version>-live-rootfs.<architecture>.img" \// <3>
+   --extra-args "ip=<ip>::<gateway>:<mask>:<hostname>::none" \ <4>
+   --extra-args "nameserver=<dns>" \
    --extra-args "random.trust_cpu=on rd.luks.options=discard" \
```
Comment (smolin-de) [inline modules/install-sno-ibm-z-kvm.adoc:167]: Why do we need these parameters ?

File: modules/install-sno-ibm-z.adoc [hunk 3]
```diff
@@ -150,20 +150,20 @@ The `rootfs` image is the same for FCP and DASD.
...
 rd.znet=qeth,0.0.bdd0,0.0.bdd1,0.0.bdd2,layer2=1 \
 rd.dasd=0.0.4411 \// <4>
 rd.zfcp=0.0.8001,0x50050763040051e3,0x4000406300000000 \// <5>
 zfcp.allow_lun_scan=0 \
 rd.luks.options=discard
```
Comment (smolin-de) [inline modules/install-sno-ibm-z.adoc:163]: Why do we need this parameter ?

Comment (SNiemann15) [reply modules/install-sno-ibm-z.adoc:163]: @chidanandbh Can you pleases take a look. Original input came from you when we added support for SNO

**Review by smolin-de — CHANGES_REQUESTED** [review]
.

File: modules/install-sno-ibm-z-kvm.adoc [hunk 4]
```diff
@@ -159,15 +159,16 @@ $ virt-install \
...
-   --extra-args "nameserver=<name_server>" \
-   --extra-args "ip=dhcp rd.neednet=1 ignition.platform.id=metal ignition.firstboot" \
-   --extra-args "coreos.live.rootfs_url=<rhcos_liveos>" \// <2>
-   --extra-args "ignition.config.url=<rhcos_ign>" \// <3>
+   --extra-args "rd.neednet=1 ignition.platform.id=metal ignition.firstboot" \
+   --extra-args "ignition.config.url=http://<http_server>/bootstrap.ign" \// <2>
+   --extra-args "coreos.live.rootfs_url=http://<http_server>/rhcos-<version>-live-rootfs.<architecture>.img" \// <3>
+   --extra-args "ip=<ip>::<gateway>:<mask>:<hostname>::none" \ <4>
+   --extra-args "nameserver=<dns>" \
    --extra-args "random.trust_cpu=on rd.luks.options=discard" \
```
Comment (smolin-de) [inline modules/install-sno-ibm-z-kvm.adoc:167]:
```suggestion
   --extra-args "nameserver=<dns>"
 ```

File: modules/install-sno-ibm-z.adoc [hunk 5]
```diff
@@ -150,20 +150,20 @@ The `rootfs` image is the same for FCP and DASD.
...
-ignition.config.url=<rhcos_ign>:8080/ignition/bootstrap-in-place-for-live-iso.ign \// <2>
-ip=encbdd0:dhcp::02:00:00:02:34:02 <3>
+ignition.config.url=http://<http_server>:8080/ignition/bootstrap-in-place-for-live-iso.ign \// <1>
+coreos.live.rootfs_url=http://<http_server>/rhcos-<version>-live-rootfs.<architecture>.img \// <2>
+ip=<ip>::<gateway>:<mask>:<hostname>::none nameserver=<dns><3>
 rd.znet=qeth,0.0.bdd0,0.0.bdd1,0.0.bdd2,layer2=1 \
 rd.dasd=0.0.4411 \// <4>
 rd.zfcp=0.0.8001,0x50050763040051e3,0x4000406300000000 \// <5>
 zfcp.allow_lun_scan=0 \
 rd.luks.options=discard
```
Comment (smolin-de) [inline modules/install-sno-ibm-z.adoc:163]:
```suggestion
zfcp.allow_lun_scan=0
```

File: modules/ibmz-configure-nbde-with-static-ip.adoc [hunk 6]
```diff
@@ -120,38 +120,36 @@ ifndef::ibm-z-kvm[]
...
-zfcp.allow_lun_scan=0 \// <4>
+coreos.inst.ignition_url=http://<http_server>/master.ign \// <2>
+coreos.live.rootfs_url=http://<http_server>/rhcos-<version>-live-rootfs.<architecture>.img \// <3>
+ip=<ip>::<gateway>:<netmask>:<hostname>::none nameserver=<dns> \
 rd.znet=qeth,0.0.bdd0,0.0.bdd1,0.0.bdd2,layer2=1 \
-rd.zfcp=0.0.5677,0x600606680g7f0056,0x034F000000000000 \// <5>
+rd.zfcp=0.0.5677,0x600606680g7f0056,0x034F000000000000 \// <4>
+zfcp.allow_lun_scan=0
 ----
 <1> For installations on DASD-type disks, add `coreos.inst.install_dev=/dev/dasda`. Omit this value for FCP-type disks.
```
Comment (smolin-de) [inline modules/ibmz-configure-nbde-with-static-ip.adoc:132]:
```suggestion
<1> For installations on DASD-type disks, add `coreos.inst.install_dev=/dev/dasda`. For installations on FCP-type disks, use coreos.inst.install_dev=/dev/sda or /dev/disk/by-id/scsi-<serial_number>.
```

**Review by smolin-de — CHANGES_REQUESTED** [review]

**Review by smolin-de — CHANGES_REQUESTED** [review]
Please check my latest comments.

File: modules/installation-ibm-z-user-infra-machines-iso.adoc [hunk 7]
```diff
@@ -86,18 +86,19 @@ Example parameter file, `bootstrap-0.parm`, for the bootstrap machine:
...
+coreos.inst.secure_ipl \// <4>
+ip=<ip>::<gateway>:<netmask>:<hostname>::none nameserver=<dns> \
 rd.znet=qeth,0.0.bdf0,0.0.bdf1,0.0.bdf2,layer2=1,portno=0 \
-zfcp.allow_lun_scan=0 \
-rd.dasd=0.0.3490
+rd.dasd=0.0.3490 \
+zfcp.allow_lun_scan=0
 ----
-<1> Specify the location of the `rootfs` artifact for the `kernel` and `initramfs` you are booting. Only HTTP and HTTPS protocols are supported.
+<1> Specify the block device on the system to install to. For installations on DASD-type disk use `dasda`, for installations on FCP-type disks use `sda`.
```
Comment (smolin-de) [inline modules/installation-ibm-z-user-infra-machines-iso.adoc:98]: Should be equal to the above change:
```suggestion
<1> Specify the block device type. For installations on DASD-type disks, specify `/dev/dasda`. For installations on FCP-type disks, specify `/dev/sda`.
```

File: modules/ibmz-configure-nbde-with-static-ip.adoc [hunk 8]
```diff
@@ -120,38 +120,36 @@ ifndef::ibm-z-kvm[]
 ----
 rd.neednet=1 \
```
Comment (smolin-de) [inline modules/ibmz-configure-nbde-with-static-ip.adoc:121]:
```suggestion
cio_ignore=all,!condev rd.neednet=1 \
```

File: modules/ibmz-configure-nbde-with-static-ip.adoc [hunk 9]
```diff
@@ -120,38 +120,36 @@ ifndef::ibm-z-kvm[]
...
-
+<1> Specify the block device type. For installations on DASD-type disks, specify `/dev/dasda`. For installations on FCP-type disks, specify `/dev/sda`.
+<2> Specify the location of the Ignition config file. Use `master.ign` or `worker.ign`. Only HTTP and HTTPS protocols are supported.
+<3> Specify the location of the `rootfs` artifact for the `kernel` and `initramfs` you are booting. Only HTTP and HTTPS protocols are supported.
+<4> For installations on DASD-type disks, replace with `rd.dasd=0.0.xxxx` to specify the DASD device.
 endif::ibm-z-kvm[]
 ifdef::ibm-z-kvm[]
 [source,terminal]
 ----
 rd.neednet=1 \
```
Comment (smolin-de) [inline modules/ibmz-configure-nbde-with-static-ip.adoc:140]:
```suggestion
cio_ignore=all,!condev rd.neednet=1 \
```

File: modules/installation-ibm-z-user-infra-machines-iso.adoc [hunk 10]
```diff
@@ -86,18 +86,19 @@ Example parameter file, `bootstrap-0.parm`, for the bootstrap machine:
 ----
 rd.neednet=1 \
```
Comment (smolin-de) [inline modules/installation-ibm-z-user-infra-machines-iso.adoc:87]:
```suggestion
cio_ignore=all,!condev rd.neednet=1 \
```

File: modules/machine-user-infra-machines-ibm-z.adoc [hunk 11]
```diff
@@ -102,12 +102,12 @@ The following is an example parameter file, `additional-worker-dasd.parm`:
 rd.neednet=1 \
```
Comment (smolin-de) [inline modules/machine-user-infra-machines-ibm-z.adoc:102]:
```suggestion
cio_ignore=all,!condev rd.neednet=1 \
```

**Review by smolin-de — CHANGES_REQUESTED** [review]

File: modules/installation-ibm-z-kvm-user-infra-machines-iso.adoc [hunk 12]
```diff
@@ -48,14 +48,14 @@ $ qemu-img create -f qcow2 -F qcow2 -b /var/lib/libvirt/images/{source_rhcos_qem
...
+   --name <vm_name> \
+   --memory <memory_mb> \
+   --vcpus <vcpus> \
+   --disk <disk> \
    --launchSecurity type="s390-pv" \ <1>
    --import \
-   --network network={network},mac={mac} \
-   --disk path={ign_file},format=raw,readonly=on,serial=ignition,startup_policy=optional <2>
+   --network network=<virt_network_parm>,mac=<mac_address> \
+   --disk path=<ign_file>,format=raw,readonly=on,serial=ignition,startup_policy=optional <2>
```
Comment (smolin-de) [inline modules/installation-ibm-z-kvm-user-infra-machines-iso.adoc:58]: I think that <ign_file> is the filename which is specified under step 3.
Therefore, it should be mentioned below under <2>.

**Review by smolin-de — CHANGES_REQUESTED** [review]

File: modules/installation-ibm-z-kvm-user-infra-machines-iso.adoc (hunk 12 above)
Comment (SNiemann15) [reply modules/installation-ibm-z-kvm-user-infra-machines-iso.adoc:58]: The step says "create the KVM guest nodes using the Ignition file..." Therefore, I don't think we need to add that information to the callout again.

**Review by smolin-de — APPROVED** [review]

**Comment by holgwolf** [comment]
LGTM

**Review by maxwelldb — APPROVED** [review]
lgtm.

**Pull request merged**